src/framework/mpas_stream_list.F

```diff
@@ -112,21 +112,23 @@ subroutine MPAS_stream_list_insert(list, stream, ierr) !{{{
 
         if (present(ierr)) ierr = MPAS_STREAM_LIST_NOERR
 
-        nullify(stream % next)
-
         if (.not. associated(list % head)) then
             list % head => stream
         else
             node => list % head
-            do while (associated(node % next))
+            do while (associated(node))
                 if (node % name == stream % name) then
                     if (present(ierr)) ierr = MPAS_STREAM_LIST_DUPLICATE
                     LIST_ERROR_WRITE('Found duplicate item '//trim(stream % name)//' in list.')
                     return
                 end if
+                if (.not. associated(node % next)) then
+                    node % next => stream
```

Comment:
As part of 999c449c, perhaps it would have led to slightly more understandable code (in my opinion) if we had left the `node % next => stream` statement in its original location below line 130 rather than moving it here, especially considering that we `nullify(stream % next)` only after adding the stream to the list. What do you think?

Reply:
That's fine, too. Whatever we do, it would be best to have the logic all together in the same place.